pkgs/development/libraries/rapidjson/unstable.nix

```diff
@@ -30,6 +30,9 @@ stdenv.mkDerivation rec {
       url = "https://git.alpinelinux.org/aports/plain/community/rapidjson/do-not-include-gtest-src-dir.patch?id=9e5eefc7a5fcf5938a8dc8a3be8c75e9e6809909";
       hash = "sha256-BjSZEwfCXA/9V+kxQ/2JPWbc26jQn35CfN8+8NW24s4=";
     })
+
+    # Due to some update in nixpkgs, valgrind is broken here
```

Comment:
What's the symptom?

Reply:
This is the only thing it tells me.

```console
[----------] Global test environment tear-down
[==========] 464 tests from 33 test suites ran. (9980 ms total)
[  PASSED  ] 464 tests.
==719== 
==719== HEAP SUMMARY:
==719==     in use at exit: 0 bytes in 0 blocks
==719==   total heap usage: 60,407 allocs, 60,407 frees, 174,209,515 bytes allocated
==719== 
==719== All heap blocks were freed -- no leaks are possible
==719== 
==719== For lists of detected and suppressed errors, rerun with: -s
==719== ERROR SUMMARY: 989242 errors from 13 contexts (suppressed: 0 from 0)

3/3 Test #1: perftest .........................   Passed   29.85 sec

67% tests passed, 1 tests failed out of 3

Total Test time (real) =  29.85 sec

The following tests FAILED:
          3 - valgrind_unittest (Failed)
Errors while running CTest
```

Reply:
This snapshot doesn't show the errors

Reply:
I know it's been a while and we want this merged asap, but

1. Got to make sure the errors aren't real
2. In case they actually are, got to report upstream and attach a comment to the patch, explaining how long it'll have to hang around
3. In case we do want to disable the test, I wonder if we could pass something like `-E "symbol_test"` to CTest instead of applying patches

Reply:
I don't know how to check if the errors are "real" or not.
All I see is that the stable rapidjson derivation also has to skip valgrind.
From what I remember, the valgrind tests used to work, so that's why I put "# Due to some update in nixpkgs, valgrind is broken here".
I'm not too worried about rapidjson-unstable's valgrind test being skipped, as it's an unstable variant and this sort of thing happens.
If you know how to check if the errors are real or not, let me know.

Reply:
Any chance you've got the logs around?

Reply:
Those are the logs.
That's why I don't know what to do.
Everything else looks completely normal.

Reply:
I meant something like this: https://gist.github.com/SomeoneSerge/23334f258b1dc58d4bef3e0095174428 (`nix log /nix/store/wnlis3xdzsaab263wzh6r38x79w4g7mq-rapidjson-1.1.0.drv | gh gist create`)

Reply:
I think I see what you mean now, and after inspecting it I wonder if these lines are the issue.
```console
==719== Source and destination overlap in memcpy_chk(0x1ffeffe352, 0x1ffeffe351, 7)
==719==    at 0x484D592: __memcpy_chk (in /nix/store/x8bsr30scw226s3w74frdrpvfg637pwl-valgrind-3.21.0/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==719==    by 0x67A098: dtoa_normal_Test::TestBody() (in /build/source/build/bin/unittest)
==719==    by 0x48AFCAC: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /nix/store/jswc4nv9v8n64pcb5mljjf3g0cjv0fm6-g>
==719==    by 0x489E47D: testing::Test::Run() (in /nix/store/jswc4nv9v8n64pcb5mljjf3g0cjv0fm6-gtest-1.12.1/lib/libgtest.so.1.12.1)
==719==    by 0x489E634: testing::TestInfo::Run() (in /nix/store/jswc4nv9v8n64pcb5mljjf3g0cjv0fm6-gtest-1.12.1/lib/libgtest.so.1.12.1)
==719==    by 0x489E754: testing::TestSuite::Run() (in /nix/store/jswc4nv9v8n64pcb5mljjf3g0cjv0fm6-gtest-1.12.1/lib/libgtest.so.1.12.1)
==719==    by 0x48A558E: testing::internal::UnitTestImpl::RunAllTests() (in /nix/store/jswc4nv9v8n64pcb5mljjf3g0cjv0fm6-gtest-1.12.1/lib/libgtest.so.1.12.1)
==719==    by 0x489E91A: testing::UnitTest::Run() (in /nix/store/jswc4nv9v8n64pcb5mljjf3g0cjv0fm6-gtest-1.12.1/lib/libgtest.so.1.12.1)
==719==    by 0x5FAB2D: main (in /build/source/build/bin/unittest)
==719==
...
```

Never used valgrind before so I didn't think that was an error before.
[log.txt](https://github.com/NixOS/nixpkgs/files/13219310/log.txt)

Reply:
It's specifically ``ParseNumber_NormalPrecisionError`` in ``readertest``.
Working on it. Might just have to strip it out or tell valgrind to ignore it though.

Reply:
@SomeoneSerge https://github.com/Tencent/rapidjson/issues/2214

Reply:
Fixed at 507def8a464c6c94cf7e82014f65fccc57f7c09b